docs/content/configuration/storage.md

```diff
@@ -335,3 +335,94 @@ It assumes that you already have the following:
             # Name of container in the blob storage
             # Env override: ATHENS_AZURE_CONTAINER_NAME
             ContainerName = "MY_AZURE_BLOB_CONTAINER_NAME"
+
+## Running multiple Athens pointed at the same storage
+
+Athens has the abilit to run concurrently pointed at the same storage medium, using
+a distributed locking mechanism called "single flight".
+
+By default, Athens is configured to use the `memory` single flight, which
+stores locks in local memory. This works when running a single Athens instance, given
+the process has access to it's own memory. However, when running multiple Athens instances
+pointed at the same storage, a distributed locking mechansism is required.
+
+Athens supports several distributed locking mechanisms:
+  - `etcd`
+  - `redis`
+  - `redis-sentinel`
+  - `gcp` (available when using the `gcp` storage type)
+  - `azureblob` (available when using the `azureblob` storage type)
+
+Setting the `SingleFlightType` (or `ATHENS_SINGLE_FLIGHT TYPE` in the environment) configuration
+value will enable usage of one of the above mechanisms. The `azureblob` and `gcp` types require
+no extra configuration.
+
+### Using etcd as the single flight mechanism
+
+Using the `etcd` mechanism is very simple, just a comma separated list of etcd endpoints.
+The recommend configuration is 3 endpoints, however, more can be used.
+  
+  SingleFlightType = "etcd"
+
+  [SingleFlight]
+      [SingleFlight.Etcd]
+          # Env override: ATHENS_ETCD_ENDPOINTS
+          Endpoints = "localhost:2379,localhost:22379,localhost:32379"
+
+### Using redis as the single flight mechanism
+
+Athens supports two mechanisms of communicating with redis: direct connection, and
+connecting via redis sentinels.
+
+#### Direct connection to redis
+
+Using a direct connection to redis is simple, and only requires a single `redis-server`.
+You can also optionally specify a password to connect to the redis server with
+
+  SingleFlightType = "redis"
+
+  [SingleFlight]
+      [SingleFlight.Redis]
+          # Endpoint is the redis endpoint for the single flgiht mechanism
+          # Env override: ATHENS_REDIS_ENDPOINT
+          Endpoint = "127.0.0.1:6379"
+
+          # Password is the password for the redis instance
+          # Env override: ATHENS_REDIS_PASSWORD
+          Password = ""
+
+#### Connecting to redis via redis sentinel
+
+**NOTE**: redis-sentinel requires a working knowledge of redis and is not recommended for
+everyone.
+
+redis sentinel is a high-availability set up for redis, it provides automated monitoring, replication,
+failover and configuration of multiple redis servers in a master-follower setup. It is more
```

Comment:
leader/follower rather than master/follower so we're not mixing leader/follower, master/slave (though to be honest either is fine just a question given might trigger master/slave issues)